wgpu-hal/src/dx12/instance.rs

```diff
@@ -20,12 +20,24 @@ impl crate::Instance<super::Api> for super::Instance {
             crate::InstanceError::with_source(String::from("failed to load d3d12.dll"), e)
         })?;
 
-        if desc.flags.contains(wgt::InstanceFlags::VALIDATION) {
+        if desc
+            .flags
+            .intersects(wgt::InstanceFlags::VALIDATION | wgt::InstanceFlags::GPU_BASED_VALIDATION)
+        {
             // Enable debug layer
             match lib_main.get_debug_interface() {
                 Ok(pair) => match pair.into_result() {
                     Ok(debug_controller) => {
-                        debug_controller.enable_layer();
+                        if desc.flags.intersects(wgt::InstanceFlags::VALIDATION) {
+                            debug_controller.enable_layer();
+                        }
+                        if desc
+                            .flags
+                            .intersects(wgt::InstanceFlags::GPU_BASED_VALIDATION)
+                            && !debug_controller.enable_gpu_based_validation()
```

Comment:
Could we keep side affects out of the if statement body?

Reply:
The reason I had combined the conditions was to resolve [the `clippy::collapsible_if` lint](https://rust-lang.github.io/rust-clippy/stable/index.html#/collapsible_if), but I agree that this just sorta makes things harder to follow. Suppressed the lint with f652549eb.